### ISSUE TITLE
Add/Create SAS permission doesn't work as expected in AzCopy

### ISSUE THREAD

**Issue by JasonYeMSFT** [Azure/azure-storage-azcopy#833]
### Which version of the AzCopy was used? 
10.3.3

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows

### What command did you run?
copy

### What problem was encountered?
AzCopy copy command automatically cancel if a user use a SAS with read, create, add, list permissions as the destination.

### How can we reproduce the problem in the simplest way?
Find a blob container and generate a SAS with "rcal" permission. Use it as destination in a copy command with a valid source. See the output.

Based on my observation, you always need a "write" permission in the destination.

### Have you found a mitigation/solution?
I would expect the "copy" with "rcal" destination to work when there is no conflict, where "write" permission is not necessary. If "write" permission is required by design, it should result in a failure and explains the reason in the log file (e.g. AzCopy reports in the log file that it cannot check for conflicts if "read" permission is missing).

**Comment by nakulkar-msft** [comment]
The error thrown is self-explainatory.

  Code: UnauthorizedBlobOverwrite
   PUT https://storenakulkar.blob.core.windows.net/container1/testcases.txt?se=2022-02-23t17%3A59%3A08z&sig=-REDACTED-&sp=rlac&spr=https&srt=sco&ss=bfqt&st=2022-02-23t09%3A59%3A08z&sv=2020-08-04&timeout=901
   Content-Length: [7686]
   User-Agent: [AzCopy/10.13.0 Azure-Storage/0.14 (go1.16; Windows_NT)]
   X-Ms-Blob-Cache-Control: []
   X-Ms-Blob-Content-Disposition: []
   X-Ms-Blob-Content-Encoding: []
   X-Ms-Blob-Content-Language: []
   X-Ms-Blob-Content-Md5: []
   X-Ms-Blob-Content-Type: [text/plain]
   X-Ms-Blob-Type: [BlockBlob]
   X-Ms-Client-Request-Id: [c8a861e0-4e32-480e-69b3-b8b0577b8305]
   X-Ms-Version: [2019-12-12]
   --------------------------------------------------------------------------------
   RESPONSE Status: 403 This request is not authorized to perform blob overwrites.
   Content-Length: [252]
   Content-Type: [application/xml]
   Date: [Wed, 23 Feb 2022 10:04:11 GMT]
   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]
   X-Ms-Client-Request-Id: [c8a861e0-4e32-480e-69b3-b8b0577b8305]
   X-Ms-Error-Code: [UnauthorizedBlobOverwrite]
   X-Ms-Request-Id: [01bbd318-201e-004c-089c-285c65000000]
   X-Ms-Version: [2019-12-12]

**Comment by JasonYeMSFT** [comment]
@nakulkar-msft I haven't verified this for a while. With the latest AzCopy, if you run a copy command with such destination which hits the problem, does it report the job as "failed" or "cancelled"? When I opened this issue, AzCopy is reporting the job as "Cancelled" which isn't very accurate.

**Comment by joshwright10** [comment]
I have just witnessed this problem when using AzcopyVersion 10.14.1.
If I upload small text files, pdf files etc. the copy is successful with the Create permission in the SAS Token. However, anything larger (this was around 100MB) and I get the following errors.

I added the Write permission to the SAS Token and the upload is successful.

> Content-Length: [8388608]
   User-Agent: [AzCopy/10.14.1 Azure-Storage/0.14 (go1.16.14; Windows_NT)]
   X-Ms-Client-Request-Id: [f1f51647-0bff-4977-4fea-1cdb46fb0bb0]
   X-Ms-Version: [2020-04-08]
   RESPONSE Status: 403 This request is not authorized to perform this operation using this permission.
   Content-Length: [279]
   Content-Type: [application/xml]
   Date: [Fri, 25 Mar 2022 09:18:33 GMT]
   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]
   X-Ms-Client-Request-Id: [f1f51647-0bff-4977-4fea-1cdb46fb0bb0]
   X-Ms-Error-Code: [AuthorizationPermissionMismatch]
   X-Ms-Request-Id: [914cbb79-d01e-005c-7529-40c6a2000000]
   X-Ms-Version: [2020-04-08]
Response Details: ﻿<Code>AuthorizationPermissionMismatch</Code><Message>This request is not authorized to perform this operation using this permission. </Message>
  2022/03/25 09:18:33 INFO: [P#0-T#0] Early close of chunk in singleChunkReader with context still active
  2022/03/25 09:18:33 ERR: [P#0-T#0] UPLOADFAILED: \\?\C:\Users\username\Downloads\filename.zip : 403 : 403 This request is not authorized to perform this operation using this permission.. When Staging block. X-Ms-Request-Id: 914cbb79-d01e-005c-7529-40c6a2000000
